### PR TITLE
Fix mongo

### LINKED ISSUE #1284: 2.13 support for mongodb
I noticed mongodb to be broken in 2.13 and saw the hiatus note on the library. I'm curious to know what specifically is broken and trying to acertain the level of effort required to fix it. Our team is willing to work on it contingent of level of effort. Any thoughts on it will help!

### PR DESCRIPTION
**Pull request by aschmois**
Before submitting your PR, check that you've:

- [X] Ran `stylish-haskell` on any changed files.
- [X] Adhered to the code style (see the `.editorconfig` file for details)

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Bumped the version number if there isn't an `(unreleased)` on the Changelog
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

Closes #1284 

Since embedded fields have been removed there are some sacrifices we had to make but the majority of the features should still be working. Embedded fields will be set as null and they will be unordered when retrieved.

### CODE REVIEW

File: persistent-mongoDB/Database/Persist/MongoDB.hs [hunk 1]
```diff
@@ -430,15 +445,18 @@ toInsertDoc record =
                 DB.:=
                     embeddedVal pv
         )
-        $ filter (\(_, pv) -> isNull pv)
+        $ filter (\(_, pv) -> not $ isNull pv)
```
Comment (aschmois) [inline persistent-mongoDB/Database/Persist/MongoDB.hs:448]: The major fix was this! At least for our needs this was the only thing that really broke anything.

Comment (parsonsmatt) [reply persistent-mongoDB/Database/Persist/MongoDB.hs:448]: Wow. I'm really surprised that worked?

**Review by parsonsmatt — CHANGES_REQUESTED** [review]
Well done! I'm surprised and impressed :joy:

Can you update the `persistent-mongoDB.cabal` file to be the new version? I can merge and release it then.

Would you be willing to accept maintainership responsibilities for `persistent-mongoDB`? I'll add you as a user on Hackage and give you access to the repository.

Thanks!

**Comment by aschmois** [comment]
> Would you be willing to accept maintainership responsibilities for `persistent-mongoDB`? I'll add you as a user on Hackage and give you access to the repository.

Sounds good. I'll have to maintain it for the time being while our team continues to use mongodb!

**Review by parsonsmatt — APPROVED** [review]
Fantastic! Thanks!

**Comment by parsonsmatt** [comment]
What's a Hackage username I can add?

**Comment by parsonsmatt** [comment]
@snoyberg Can we add @aschmois as a maintainer on the repo?

**Comment by aschmois** [comment]
I have an upload account with andres.schmois@gmail.com

**Comment by snoyberg** [comment]
No problem from my side. I just added you on Hackage for persistent-mongoDB and as a repo writer. Thank you!

**Comment by parsonsmatt** [comment]
@aschmois Want to do the honors (sqash+merge+release)?

**Pull request merged**